cluster/router/tag/router_rule.go

```diff
@@ -34,5 +52,54 @@ func getRule(rawRule string) (*RouterRule, error) {
 		return r, err
 	}
 	r.RawRule = rawRule
+	r.init()
 	return r, nil
 }
+
+// init use for flattening tags data to @addressToTagNames and @tagNameToAddresses
+func (t *RouterRule) init() {
+	t.addressToTagNames = make(map[string][]string, 2*len(t.Tags))
```

Comment:
I do not think named `init` is golang native.  Because `init` is golang's  special func. Maybe   func (t *RouterRule)NewRouterRule is .